MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/labs/IndoorMapActivity.java

```diff
@@ -0,0 +1,227 @@
+import com.mapbox.services.commons.turf.TurfJoins;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mapbox.mapboxsdk.style.layers.Function.stop;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillOpacity;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineOpacity;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineWidth;
+
+public class IndoorMapActivity extends AppCompatActivity {
+
+  private GeoJsonSource indoorBuildingSource;
+  private boolean levelButtonsShowing = true;
+  private List<Position> boundingBox;
+  private View levelButtons;
+  private MapView mapView;
+  private MapboxMap map;
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    MapboxAccountManager.start(this, getString(R.string.access_token));
+    setContentView(R.layout.activity_lab_indoor_map);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(final MapboxMap mapboxMap) {
+        map = mapboxMap;
+
+        levelButtons = findViewById(R.id.floor_level_buttons);
+
+        boundingBox = new ArrayList<>();
+        boundingBox.add(Position.fromCoordinates(-77.03791, 38.89715));
+        boundingBox.add(Position.fromCoordinates(-77.03791, 38.89811));
+        boundingBox.add(Position.fromCoordinates(-77.03532, 38.89811));
+        boundingBox.add(Position.fromCoordinates(-77.03532, 38.89708));
+
+        mapboxMap.setOnCameraChangeListener(new MapboxMap.OnCameraChangeListener() {
+          @Override
+          public void onCameraChange(CameraPosition position) {
+
+            if (position.zoom > 16) {
+              try {
+                if (TurfJoins.inside(Position.fromCoordinates(
+                  position.target.getLongitude(),
+                  position.target.getLatitude()),
+                  boundingBox)) {
+                  if (!levelButtonsShowing) {
+                    showLevelButton();
+                  }
+                } else {
+                  if (levelButtonsShowing) {
+                    hideLevelButton();
+                  }
+                }
+              } catch (TurfException turfException) {
+                turfException.printStackTrace();
+              }
+            } else if (levelButtonsShowing) {
+              hideLevelButton();
+            }
+          }
+        });
+
+        String geojsonString = loadJsonFromAsset("white_house_lvl_0.geojson");
+        if (geojsonString != null) {
+          indoorBuildingSource = new GeoJsonSource("indoor-building", geojsonString);
```

Comment:
This doesn't work as expected. As soon as you hand off the source to `addSource()` the handle becomes invalid. So either look it up here or jit when you need it.